core/templates/dev/head/pages/exploration-editor-page/services/exploration-states.service.spec.ts

```diff
@@ -144,4 +148,21 @@ describe('ExplorationStatesService', function() {
       });
     });
   });
+
+
+  it('solicit answer details value should be false', function() {
```

Comment:
Start with `should ..`. 

If you'll make it `it('should save the solicitAnswerDetails correctly', ...` then you'll see that the describe message `ExplorationStatesService ` (on line 23)   + the message for 'it` block will make a sentence.
`ExplorationStatesService should save the solicitAnswerDetails correctly`.

Inside the test, block follows this pattern:

- Do a baseline verification i.e, expect the existing value to be something(false).
- Change/save the value of `SolicitAnswerDetails` to somethingelse (true).
- Do a final verification that save function is working correctly i.e, expect the `SolicitAnswerDetails` to be somethingelse(true)


(I agree that the other tests in this file don't follow that pattern and that's why it was hard for you to write the test. (You can follow other files for help.)

Reply:
Thanks Sandeep for the explanation.